src/AbstractOperations/conditional_operations.jl

```diff
@@ -176,12 +176,26 @@ function materialize_condition(c::ConditionalOperation)
     return f
 end
 
-@inline function conditional_one(c::ConditionalOperation, mask)
+const NotWindowedField = Field{<:Any, <:Any, <:Any, <:Any, <:Any, Tuple{Colon, Colon, Colon}}
+const NotWindowedFieldCO = ConditionalOperation{<:Any, <:Any, <:Any, <:Any, <:Any, <:NotWindowedField}
+
+# if not a windowed field, use OneField for performance
+@inline function conditional_one(c::NotWindowedFieldCO, mask)
     LX, LY, LZ = location(c)
     one_field = OneField(Int)
     return ConditionalOperation{LX, LY, LZ}(one_field, nothing, c.grid, c.condition, mask)
 end
 
+# for windowed fields we need to pass indices
+@inline function conditional_one(c::ConditionalOperation, mask)
+    LX, LY, LZ = location(c)
+    grid = c.operand.grid
+    indices = c.operand.indices
+    one_field = Field{LX, LY, LZ}(grid; indices)
+    set!(one_field, 1)
```

Comment:
```suggestion
    if indices == (:, :, :)
        one_field = OneField(Int)
    else
        one_field = Field{LX, LY, LZ}(grid; indices)
        set!(one_field, 1)
    end
```

Reply:
@glwagner I essentially already did what you suggest but via dispatch; see: https://github.com/CliMA/Oceananigans.jl/blob/50131275c1e7ae344abcdfd3933bab423f677a09/src/AbstractOperations/conditional_operations.jl#L179-L197

Did you miss that? Or there is a way to prefer using `if else`?

Reply:
provided that type inference succeeds, the if-statement is identical to manual dispatch in terms of generated code

Reply:
so its up to you which you find to be easier to read and maintain

Reply:
(from Slack chat)

glwagner: the two are identical because the type of `indices` is known. so its up to you in terms of readability and maintainability, i think the if statement is a little clearer to read

Reply:
agree, if is cleaner!